tests/fixtures/e2e_suite.go

```diff
@@ -82,6 +84,61 @@ func (s *E2ESuite) TearDownSuite() {
 	close(s.stopch)
 }
 
+func (s *E2ESuite) BeforeTest(suiteName, testName string) {
+	// ensure local repo has been deleted incase previous test run failed
+	err := os.RemoveAll(localPath)
+	s.CheckError(err)
+}
+
+func (s *E2ESuite) AfterTest(suiteName, testName string) {
+
+	// open local path to cloned git server
+	repo, err := git.PlainOpen(localPath)
+	if err != nil {
+		s.CheckError(err)
+	}
+
+	// open worktree
+	wt, err := repo.Worktree()
+	if err != nil {
+		s.CheckError(err)
+	}
+
+	// find path to test repo in local
+	entries, err := os.ReadDir(localPath)
+	if err != nil {
+		s.CheckError(err)
+	}
+
+	for _, entry := range entries {
+		// clean out repo of all changes
+		if entry.IsDir() {
+			_, err = wt.Remove(entry.Name())
+			if err != nil {
+				s.CheckError(err)
+			}
+		}
+	}
+
+	_, err = wt.Commit("Cleaning out repo", &git.CommitOptions{})
+	if err != nil {
+		s.CheckError(err)
+	}
+
+	// git push to remote
+	err = repo.Push(&git.PushOptions{
+		RemoteName: "origin",
+		Auth:       auth,
+	})
+	if err != nil {
+		s.CheckError(err)
+	}
```

Comment:
nit: can you break this out into a function called `ResetRepository()` or something?